cmds/cksum/cksum.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+func helpPrinter() {
+	fmt.Printf("Usage:\ncksum <File Name>\n")
+	flag.PrintDefaults()
+	os.Exit(0)
+}
+
+func versionPrinter() {
+	fmt.Println("cksum utility, URoot Version.")
+	os.Exit(0)
+}
+
+func GetInput(fileName string) (input []byte, err error) {
+	if fileName != "" {
+		return ioutil.ReadFile(fileName)
+	}
+	return ioutil.ReadAll(os.Stdin)
+}
+
+func printCksum(input []byte) string {
+
+	var cksumTable = []uint64{
```

Comment:
Why use uint64s for everything if your end result is a 32 bit CRC?

Reply:
Some intermediate value seems to be overflowing with int.